apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/shard/observer/ShardObserver.java

```diff
@@ -59,17 +59,31 @@ public void onTrimDone(@Observes @Sync TrimData trimData) {
         boolean isShardingOff = propertiesHolder.getBooleanProperty("apl.noshardcreate", false);
         boolean shardingEnabled = currentConfig.isShardingEnabled();
         log.debug("Is sharding enabled ? : '{}' && '{}'", shardingEnabled, !isShardingOff);
+        int shardingFrequency = 1;
         if (shardingEnabled && !isShardingOff) {
+            if (!blockchainConfig.isJustUpdated()) {
+                // config didn't change from previous trim scheduling
+                shardingFrequency = currentConfig.getShardingFrequency();
+            } else {
+                // config has changed from previous trim scheduling, try to get previous 'shard frequency' value
+                shardingFrequency = blockchainConfig.getPreviousConfig().isPresent() ?
+                        blockchainConfig.getPreviousConfig().get().getShardingFrequency() // previous config
```

Comment:
sharding should be enabled in prev config

Reply:
Added check in previous config settings